lock-impl/src/main/java/com/palantir/lock/impl/LegacyTimelockService.java

```diff
@@ -186,18 +173,72 @@ public long currentTimeMillis() {
 
     @Override
     public TimestampLeaseResults acquireTimestampLeases(Map<TimestampLeaseName, Integer> requests) {
-        // TODO(aalouane): implement!
-        throw new UnsupportedOperationException("Not implemented yet!");
+        // This implementation here treats ALL the timestamp leases the same, meaning they will all progress
+        // at the same rate. This is only a perf concern (all timestamps progress together), not correctness.
+        // TODO(jakubk): This should reject timestamp lease names that conflict with the immutable timestamp lease
+        // or we should fix the implementation in timelock to allow these to conflict.
```

Comment:
One thing to add is that when doing named lease things, it WONT intersect with immutable timestamp things.

Alternatives I've thought about that might be better, but feels like your implementation is the right spot:

- Use a different client ID for each timestamp name. This risks higher probability of collision with user lock clients IDs. If we take into account how long before lockV1 gets deprecated + how many timestamp lease names we want to eventually use, it might not be too bad to just use a distinct id for each lease name.

- The better version of ^ is to outright add new APIs in the lock V1 service that specifically add namespacing. This would be using some namespaced version of `ClientId` in `versionMap` in `LockServiceImpl`. This is a bit annoying, the code there is quite legacy and we want to deprecate eventually so the lift may not be worth the improvement.

>         // TODO(jakubk): This should reject timestamp lease names that conflict with the immutable timestamp lease
>         // or we should fix the implementation in timelock to allow these to conflict.

I don't understand this one. What do you mean by conflict here?

Reply:
I think this implementation is good enough for now, we may revisit.

> I don't understand this one. What do you mean by conflict here?

I think in Timelock, you made it so that the TimestampLeaseName cannot have a "ImmutableTs" prefix. I think we should make sure we either fix this in timelock, so the names can be whatever you want, or enforce it everywhere.

Reply:
Aaah. The protection is inside `TimestampLeaseName` and done before creation, so can't get an instance where it starts with "ImmutableTimestamp"